### PR TITLE
Added:

### PR DESCRIPTION
**Pull request by Vosker**
-login.jsp + login logic + mapping
-findByLogin methods to dao and services
-AuthenticationFilter + AuthenticationService

### CODE REVIEW

**Review by voloshynamaryna11 — COMMENTED** [review]
good job

File: src/main/java/com/internet/shop/controllers/users/LoginController.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        try {
+            User user = authenticationService.login(login, password);
+            HttpSession session = req.getSession();
+            session.setAttribute(USER_ID, user.getId());
+        } catch (AuthenticationException e) {
+            req.setAttribute("errorMessage", e.getMessage());
+            req.getRequestDispatcher("/WEB-INF/views/login.jsp").forward(req, resp);
+            return;
+        }
+        resp.sendRedirect(req.getContextPath() + "/");
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/controllers/users/LoginController.java:40]:
```suggestion
        try {
            User user = authenticationService.login(login, password);
            HttpSession session = req.getSession();
            session.setAttribute(USER_ID, user.getId());
            resp.sendRedirect(req.getContextPath() + "/");
        } catch (AuthenticationException e) {
            req.setAttribute("errorMessage", e.getMessage());
            req.getRequestDispatcher("/WEB-INF/views/login.jsp").forward(req, resp);
        }
```

File: src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import com.internet.shop.service.UserService;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String login, String password) throws AuthenticationException {
+        final String error = "Incorrect login or password";
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java:16]: let's make it constant

Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java:16]:
```suggestion
        final String exceptionMessage = "Incorrect login or password";
```

**Review by dmytro-hryhoruk — APPROVED** [review]
Good job

**Pull request merged**